src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperContentExtensions.cs

```diff
@@ -51,7 +52,18 @@ public static class TagHelperContentExtensions
 
```

Comment:
Realllly dislike this extension method. I'd like to nuke the method and inline this behavior in the `Script` and `Link` `TagHelper`s (method is super misleading as is).

Copied the below uglyness (`StringWriter` stuff) from `RazorPage` because it doesn't need to be first in the page anymore.